packages/opentelemetry-instrumentation-http/src/http.ts

```diff
@@ -392,7 +393,9 @@ export class HttpInstrumentation extends InstrumentationBase<Http> {
             )
         )
       ) {
-        return original.apply(this, [event, ...args]);
+        return context.with(suppressInstrumentation(context.active()), () => {
+          return original.apply(this, [event, ...args]);
```

Comment:
The code path where a span is created binds the context to request and response objects (which are EventEmitters).
Shouldn't we do the same here?

Reply:
Not really, we only want to use `supressInstrumentation` when the request path is ignored from the config (condition above), the rest is normal execution where the request should be traced

Reply:
Maybe I was unclear.
consider following sample and we want to suppress request to `/health`:
```
function (req, res) {
  if (req.url === '/health') {
    myTracer.startSpan();  // this one is suppressed now
    req.on('data', () => myTracer.startSpan()); // should we suppress also this one as it is still the /health request?
    req.on('end', () => myTracer.startSpan()); // should we suppress also this one as it is still the /health request?
    res.on() ...
    return;
  }
  ...
}
```
For the non suppress case the `.bind` calls ensure that we link the spans correct.
Therefore I guess we the `.bind` here also to correctly suppress.

Reply:
Yeah i see, i've added the bind for `request` and `response`